english-quest-reborn/src/features/auth/auth.service.js

```
@@ -63,304 +50,197 @@ export async function initializeAuth() {
-      // Récupérer le profil actuel pour vérifier le nom d'utilisateur
-      const currentProfile = await getDocument(collections.PROFILES, authState.user.uid);
-
-      // Seul Ollie peut être administrateur, et on ne peut pas lui retirer ce droit
-      if (currentProfile && currentProfile.username && currentProfile.username.toLowerCase() === 'ollie') {
-        // Forcer isAdmin à true pour Ollie
-        profileDataToUpdate.isAdmin = true;
-        console.log("Protection des privilèges administrateur pour Ollie");
-      } else {
-        // Supprimer la tentative de modification des droits d'administrateur
+    if ('isAdmin' in profileDataToUpdate && currentProfileUsername?.toLowerCase() !== 'ollie') {
         delete profileDataToUpdate.isAdmin;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)